python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +724,37 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_to_check = config_file.get("check_ext", {})
+    # Fix the paths (OS-dependent)
+    for path, exts in dirs_to_check.items():
+        fixed_path = os.path.join(*path.split('/'))
+        config['check_ext'][fixed_path] = exts
+
     # Override default configs
     user_configs = config_file.get("configs", [])
     for pref in user_configs:
         if pref in config:
             config[pref] = user_configs[pref]
 
 
+def check_directory_files(only_changed_files, print_text=True):
+    if print_text:
+        print '\rChecking directories for correct file extensions...'
+    for directory, file_extensions in config['check_ext'].items():
+        files = get_file_list(directory, only_changed_files, config["ignore"]["directories"])
```

Comment:
`get_file_list` is useful for traversing through a folder structure. But, we want to make sure that a directory contains only 'files' of certain extensions. I think `os.listdir` would be simpler and would fit best for our purpose. That way, we can also enforce that a directory only contains the desired files, and doesn't contain any sub-dirs. And, we don't want the "ignore" part, since we won't be having any sub-dirs in these dirs.